### PR TITLE
field() function like in CakePHP 2.x

### PR DESCRIPTION
**Pull request by markusramsak**
field() function like in CakePHP 2.x

### CODE REVIEW

**Comment by dereuromark** [comment]
Is conditions and order really the only thing it needs?
I use a single $options argument instead: https://github.com/dereuromark/cakephp-shim/blob/3.0/src/Model/Table/Table.php#L195
As it might need more keys for the finder, and as such your approach limits it a little bit too much maybe.

File: src/ORM/Table.php [hunk 1]
```diff
@@ -1048,6 +1048,27 @@ protected function _setFieldMatchers($options, $keys)
...
+     * get value of a single column
+     * 
+     * @param string|array $field string field or key value pair array
+     * @param array $conditions
+     * @param array $order
+     * @return mixed|null null field value or null is returned
+     */
+    public function field($field, $conditions = [], $order = [])
+    {
+        $result = $this->find()->select(is_array($field) ? $field : [$field])->where($conditions)->order($order)->first();
```
Comment (dereuromark) [inline src/ORM/Table.php]: You can also just cast instead of `is_array($field) ? $field : [$field]`:

     (array)$field

**Comment by lorenzo** [comment]
I'm :-1: on this, the method is too limited in terms of what the query can accept.

**Comment by mdlouhy** [comment]
I used it for things like
```
$nextNumber = 1 + $this->Orders->field(['number' => 'max(number)'], ['deleted_date IS' => null]);
```
or
```
$lastModifiedDate = $this->Orders->field('last_modified_date', ['id' => $id]);
```
of course you can do this with the find() method pretty easy too, but I had use cases like this pretty often therefore I think that this method is useful.

**Comment by jhli** [comment]
You can easily implement this behaving exactly like find yet auto executing projection on that field and returning that fields value in your AppTable? Also it coulddo something like hasField checks and throw exceptions in case its not there.

**Comment by markstory** [comment]
I am a :-1: on this as well. You can use the query builder methods to do this. Finders are also a good way to encapsulate these types of operations.

**Comment by dereuromark** [comment]
I also use it mainly as BC wrapper in the Shim plugin - to ease upgrading 2.x apps. Other than that this is best left as user land implementation.

**Pull request closed without merging**